From ed5469655a4f9188a6c6963591d809f1e76b8350 Mon Sep 17 00:00:00 2001
From: Andrew Lewis <andrewlewis@google.com>
Date: Sun, 31 Jan 2021 16:53:51 +0000
Subject: [PATCH] Add a null check in RTSPSource::stop

This was causing a null pointer dereference in stop.

Manual backport of aosp/1456267.

Bug: 170643945
Bug: 177500688
Test: backport of aosp/1456267, tested in CTS
Change-Id: I8c771cd68663ebb9df79584ba5dd4edb6636bdbd
---
 media/libmediaplayerservice/nuplayer/RTSPSource.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
index 83da092506..9533ae5279 100644
--- a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
+++ b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
@@ -146,7 +146,9 @@ void NuPlayer::RTSPSource::stop() {
     }
 
     // Close socket before posting message to RTSPSource message handler.
-    close(mHandler->getARTSPConnection()->getSocket());
+    if (mHandler != NULL) {
+        close(mHandler->getARTSPConnection()->getSocket());
+    }
 
     sp<AMessage> msg = new AMessage(kWhatDisconnect, this);
 
-- 
2.29.0

